### ISSUE TITLE
Bug with datetime serielization

### ISSUE THREAD

**Issue by sfc-gh-pkommini** [Snowflake-Labs/geff#3]
Got the following error while trying the boto3 driver:

```sql
CREATE OR REPLACE EXTERNAL FUNCTION db.schema.boto3_list_repos()
RETURNS VARIANT
API_INTEGRATION="api_integation_name"
HEADERS = (
    'client-name' = 'ecr'
    'method-name' = 'describe_repositories'
    'region'      = 'us-west-2'
)
AS 'https://abc12345.execute-api.us-west-2.amazonaws.com/dev/boto3';
```

```
[ERROR] TypeError: Object of type datetime is not JSON serializable
Traceback (most recent call last):
  File "/var/task/geff/lambda_function.py", line 203, in lambda_handler
    return sync_flow(event, context)
  File "/var/task/geff/lambda_function.py", line 152, in sync_flow
    data_dumps = dumps({'data': res_data})
  File "/var/lang/lib/python3.8/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/var/lang/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/var/lang/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/var/lang/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
```

**Comment by sfc-gh-afedorov** [comment]
SnowAlert has a pretty generic solution implemented in `utils.json_dump(...)` [here](https://github.com/snowflakedb/SnowAlert/blob/master/src/runners/utils.py#L35).

**Comment by sfc-gh-pkommini** [comment]
With the goal to minimize code I generally do this to be generic enough for all python objects that usually have an __str__ method implemented:

```pyhon
Python 3.8.11 (default, Jun 29 2021, 03:08:07) 
[Clang 12.0.5 (clang-1205.0.22.9)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from datetime import datetime
>>> import json
>>> a = {'now': datetime.now()}
>>> a
{'now': datetime.datetime(2021, 8, 4, 8, 54, 44, 924393)}
>>> json.dumps(a, default=str)
'{"now": "2021-08-04 08:54:44.924393"}'
>>> 
```

**Comment by sfc-gh-pkommini** [comment]
@sfc-gh-afedorov Do you think this serves the purpose or would like me to implement the whole custom function from SnowAlert?

**Comment by sfc-gh-pkommini** [comment]
Fixed by #5
